go/pkg/log/server/property_test.go

```diff
@@ -0,0 +1,158 @@
+		id := rapid.String().Draw(t, "record_id")
+		return &coordinatorpb.OperationRecord{
+			Id: id,
+			Vector: &coordinatorpb.Vector{
+				Vector: data,
+			},
+		}
+	}))
+	rapid.Check(suite.t, func(t *rapid.T) {
+		t.Repeat(map[string]func(*rapid.T){
+			"pushLogs": func(t *rapid.T) {
+				c := collectionGen.Draw(t, "collection")
+				records := recordGen.Draw(t, "record")
+				r, err := suite.logServer.PushLogs(ctx, &logservicepb.PushLogsRequest{
+					CollectionId: c.String(),
+					Records:      records,
+				})
+				if err != nil {
+					t.Fatal(err)
+				}
+				if int32(len(records)) != r.RecordCount {
+					t.Fatal("record count mismatch", len(records), r.RecordCount)
+				}
+				suite.model.CollectionData[c] = append(suite.model.CollectionData[c], records...)
+			},
+			"getAllCollectionsToCompact": func(t *rapid.T) {
+				result, err := suite.logServer.GetAllCollectionInfoToCompact(ctx, &logservicepb.GetAllCollectionInfoToCompactRequest{})
+				assert.NoError(suite.t, err)
+				for _, collection := range result.AllCollectionInfo {
+					id, err := types.Parse(collection.CollectionId)
+					if err != nil {
+						t.Fatal(err)
+					}
+					compactionOffset := rapid.Int64Range(suite.model.CollectionCompactionOffset[id], int64(len(suite.model.CollectionData))).Draw(t, "new_position")
+					_, err = suite.logServer.UpdateCollectionLogOffset(ctx, &logservicepb.UpdateCollectionLogOffsetRequest{
+						CollectionId: id.String(),
+						LogOffset:    compactionOffset,
+					})
+					if err != nil {
+						t.Fatal(err)
+					}
+					suite.model.CollectionCompactionOffset[id] = compactionOffset
+				}
+			},
+			"pullLogs": func(t *rapid.T) {
+				c := collectionGen.Draw(t, "collection")
+				startOffset := rapid.Int64Range(suite.model.CollectionCompactionOffset[c], int64(len(suite.model.CollectionData))).Draw(t, "start_offset")
+				// If start offset is 0, we need to set it to 1 as the offset is 1 based
+				if startOffset == 0 {
+					startOffset = 1
+				}
+				batchSize := rapid.Int32Range(1, 20).Draw(t, "batch_size")
+				response, err := suite.logServer.PullLogs(ctx, &logservicepb.PullLogsRequest{
+					CollectionId:    c.String(),
+					StartFromOffset: startOffset,
+					BatchSize:       batchSize,
+					EndTimestamp:    time.Now().Unix(),
+				})
+				if err != nil {
+					t.Fatal(err)
+				}
+				// Verify that record returned is matching the expected record
+				for _, record := range response.Records {
+					expectedRecord := suite.model.CollectionData[c][record.LogOffset-1]
+					if string(expectedRecord.Vector.Vector) != string(record.Record.Vector.Vector) {
+						t.Fatalf("expect record vector %s, got %s", string(expectedRecord.Vector.Vector), string(record.Record.Vector.Vector))
+					}
+					if expectedRecord.Id != record.Record.Id {
+						t.Fatalf("expect record id %s, got %s", expectedRecord.Id, record.Record.Id)
+					}
+				}
+
+				// Verify that the first and last record offset is correct
```

Comment:
Can we verify the offset in the middle of the batch as well?

Reply:
if the number count and the first and last offset is correct, I expect the offset to be correct for all of them.